### ISSUE TITLE
No messages are stored with mod_mam activated

### ISSUE THREAD

**Issue by mapama** [processone/ejabberd#1027]
I am using ejabberd 16.02 from jessie backports and try to use mod_mam, activated like this:
```
mod_mam:
    iqdisc: one_queue
    db_type: mnesia
    default: always
```
But no messages are stored. I checked this by looking at the archive_msg entry on the webinterface and also tried to fetch the messages with gajim 0.16.5 and conversations 1.11.1.
I have absolutely no idea what's going on and why it doesn't work correctly.

Thanks very much for help.

Patrik

**Comment by Nitin3589** [comment]
Change your   db_type and it will work .
db_type : odbc

**Comment by mremond** [comment]
Note: Web interface only show data in internal database.  Moreover, from what you pasted here, indentation of your file seems incorrect.

**Comment by mapama** [comment]
Thanks for your answers.
@Nitin3589 did I miss something? If I understood it right using odbc as db_type requires me to configure a supported database while mnesia is using the internal database. Therefore I thought it must be visible in the webinterface. @mremond you are right, the indentation shown above is wrong. In my config it should be right. I think markdown eats the leading spaces for the last three lines.

Using db_type : odbc without configuring more doesn't work.

I would be very happy for more clarifying information. Thank you very much in advance.

**Comment by Nitin3589** [comment]
Yes you need to configure support database ,as your internal database has storage limits .

**Comment by mapama** [comment]
Thanks @Nitin3589. So do I understand this right, using the internal database via db_type : mnesia it is not possible to store messages via mam, so that I have to use an external database.

**Comment by mremond** [comment]
@mapama No, you can store messages in internal database. If you have large volumes, you will likely need an external database, but both should work.

**Comment by mapama** [comment]
Hi @mremond I tried it again with the internal database and db_type : mnesia, no errors in the log file and the client (conversations) tells me that mam is available at the server but it still doesn't work.

**Comment by mapama** [comment]
Can anyone give me a hint how to find the problem? Thanks very much.

Patrik

**Comment by badlop** [comment]
It is a strange problem, because from what you say, it should work. I'll describe how I tested, maybe it gives you some hint what's your problem:

I installed ejabberd git master, changed the config like you:
```
modules:
  mod_mam:
    iqdisc: one_queue
    db_type: mnesia
    default: always
...
```
Then created two accounts, user1@localhost and user2@localhost. Logged in to the first one, and sent this stanza:
```
<message to='user2@localhost'	type='chat'>
  <body>Hi bro</body>
</message>
```
Then, in the WebAdmin page https://localhost:5280/admin/node/ejabberd@localhost/db// saw that the two messages are stored (one because user1 is sender, and other because user2 is receiver):
> archive_msg	2	6,346

I tested with ejabberd git master, because I see there aren't changes in mod_mam.erl since 16.02 in it.
